### PR TITLE
Improvement/searchpage

### LINKED ISSUE #16: Fix search routing and database connection
When you try to access this page it still gives an error.
The routing and database connection needs to be set up for this page.

### PR DESCRIPTION
**Pull request by NathanNeelis**
## Fixed the following:
* added a route to /search
* Removed the ES-lint disable from the homepage (still had that on there)
* Added the function search which gathers the data from the database
* Updated some EJS templating so they implement the right data
* Added error handling

### Still to do:
I am going to make the filter function work on this page.  
I will be doing this in a separate branch as a separate feature.
Also what would be a 'nice to have' is that the person's own profile won't be in the search results.

### Note: 
Apparently I have to use an async function to combine a constructor to a database function to make it work. That's the reason I wrote an async/ await function.

### Linked issues
closes #16

### CODE REVIEW

File: src/router.js [hunk 1]
```diff
@@ -7,21 +7,22 @@ const
...
 
 const userRedirectLogin = (req, res, next) => {
-	if (!req.session.sessionID) {
-		res.redirect('/login');
-	} else {
-		next();
+    if (!req.session.sessionID) {
+        res.redirect('/login');
+    } else {
+        next();
```
Comment (BVictorB) [inline src/router.js]: You replaced the tabs with spaces (which should have resulted in warnings from eslint). Please replace these with tabs to make sure we don't keep changing these lines with every merge.

File: src/router.js [hunk 2]
```diff
@@ -7,21 +7,22 @@ const
...
 
 const userRedirectProfile = (req, res, next) => {
-	if (req.session.sessionID) {
-		res.redirect('/profile');
-	} else {
-		next();
+    if (req.session.sessionID) {
+        res.redirect('/profile');
+    } else {
+        next();
```
Comment (BVictorB) [inline src/router.js]: Same as for the lines 14-17, the spaces should be tabs!

File: src/routes/search.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            }
+        };
+        done(allData, user);
+    }
+    catch (err) {
+        res.send('something went wrong in the gathering the data');
+    }
+};
+
+module.exports = search;
```
Comment (BVictorB) [inline src/routes/search.js:30]: Looking good, nice function, clean code.

File: src/views/includes/section_result.ejs [hunk 4]
```diff
@@ -6,7 +6,7 @@
         <% for (let i = 0; i < data.length; i++) { %>
         <article>
             <a href="/<%= data[i].profileId %>">
-                <img src="uploads/<%= data[i].profilePicture %>" alt="Profile photo">
+                <img src="<%= data[i].avatar %>" alt="Profile photo">
```
Comment (BVictorB) [inline src/views/includes/section_result.ejs:9]: Perfect! It should display the users' avatar now!

File: src/router.js (hunk 1 above)
Comment (NathanNeelis) [reply src/router.js]: I don't think so actually. This is what I slacked you earlier about that my "beautify" indent was different. But we came to the conclusion that your indent here might not have been correct. And I saw this change in a pull request of yours earlier as well, so it should actually be the same now.  
  
I think this is also the reason it says "outdated" on your comment :)

File: src/router.js (hunk 2 above)
Comment (NathanNeelis) [reply src/router.js]: I don't think so actually. This is what I slacked you earlier about that my "beautify" indent was different. But we came to the conclusion that your indent here might not have been correct. And I saw this change in a pull request of yours earlier as well, so it should actually be the same now.  
  
I think this is also the reason it says "outdated" on your comment :)

File: src/routes/search.js (hunk 3 above)
Comment (NathanNeelis) [reply src/routes/search.js:30]: Cheers! Learned from the master 😃

File: src/views/includes/section_result.ejs (hunk 4 above)
Comment (NathanNeelis) [reply src/views/includes/section_result.ejs:9]: Indeed it does 🕵️

File: src/router.js (hunk 1 above)
Comment (NathanNeelis) [reply src/router.js]: If you agree, please resolve the conversation

File: src/router.js (hunk 2 above)
Comment (NathanNeelis) [reply src/router.js]: If you agree, please resolve the conversation

File: src/router.js (hunk 1 above)
Comment (BVictorB) [reply src/router.js]: Not sure what happened, it does indeed look different compared to when I reviewed it.

File: src/router.js (hunk 2 above)
Comment (BVictorB) [reply src/router.js]: Not sure what happened, it does indeed look different compared to when I reviewed it.

**Pull request merged**